src/KeywordCalls.jl

```diff
@@ -182,8 +182,8 @@ function _kwalias(__module__, fsym, aliasmap)
 end
 
 # See https://github.com/cscherrer/KeywordCalls.jl/issues/22
-@inline instance_type(f::F) where {F<:Function} = F
-@inline instance_type(f::UnionAll) = Type{f}
-
+@inline instance_type(::F) where {F} = F
+@inline instance_type(::UnionAll) = Type{<:T}
+@inline instance_type(::DataType) = Type{T}
```

Comment:
Depending on how they're used, this and the line above might need to check for `Base.has_free_typevars` first. Otherwise you might run into situations where `x isa instance_type(x)` doesn't hold:

```julia
julia> x = Vector{Core._typevar(:T, Union{}, Any)}
Array{T, 1}

julia> x isa DataType
true

julia> x isa Type{x}
false
```

If that's not the case, you could just replace this entire function with `Core.Typeof`.

Reply:
Interesting, I've never thought of this case. It's a weird corner case, since
```julia
julia> x = Vector{Core._typevar(:T, Union{}, Any)}
Array{T, 1}

julia> x isa UnionAll
false

julia> x isa Type{<:x}
false
```
I'll need to think some more about whether this case comes up. And thanks for the pointer to `Core.Typeof`, I hadn't seen that before.

Reply:
Looking at the code, it does seem like it's used just like `Core.Typeof`, so should be fine to replace it with that

Reply:
Changing `instance_type` calls to `Core.Typeof` makes tests fail, with
```julia
ERROR: LoadError: cannot assign variables in other modules
Stacktrace:
 [1] setproperty!(x::Module, f::Symbol, v::Function)
   @ Base ./Base.jl:32
 [2] _kwcall(__module__::Module, ex::Expr)
   @ KeywordCalls ~/git/KeywordCalls.jl/src/KeywordCalls.jl:55
 [3] var"@kwcall"(__source__::LineNumberNode, __module__::Module, ex::Any)
   @ KeywordCalls ~/git/KeywordCalls.jl/src/KeywordCalls.jl:37
 [4] include(fname::String)
   @ Base.MainInclude ./client.jl:476
 [5] top-level scope
   @ none:6
in expression starting at /home/chad/git/KeywordCalls.jl/test/runtests.jl:7
in expression starting at /home/chad/git/KeywordCalls.jl/test/runtests.jl:7
ERROR: Package KeywordCalls errored during testing
```